lib/bitcoin/SidetreeTransactionParser.ts

```diff
@@ -121,11 +123,11 @@ export default class SidetreeTransactionParser {
   private async fetchOutput (transactionId: string, outputIndexToFetch: number): Promise<BitcoinOutputModel | undefined> {
     try {
       const transaction = await this.bitcoinClient.getRawTransaction(transactionId);
-
       return transaction.outputs[outputIndexToFetch];
     } catch (e) {
       Logger.warn(`Error while trying to get outputIdx: ${outputIndexToFetch} from transaction: ${transactionId}. Error: ${SidetreeError.stringify(e)}`);
-      return undefined;
+      EventEmitter.emit(EventCode.BitcoinTransactionParserFailure);

```

Comment:
I'd argue that there is no need in throwing an event here, since a generic transaction processing failure event in the outer loop will be emitted up on this throw. If we really want this info to be doubled up, I'd say the thing to do is to throw a specific `SidetreeError`, so the error code is included as part of the failure label.